src/engine/InputQuery.cpp

```diff
@@ -290,26 +290,51 @@ void InputQuery::saveQuery( const String &fileName )
 
     // Number of variables
     queryFile->write( Stringf( "%u\n", _numberOfVariables ) );
+
     // Number of Bounds
     queryFile->write( Stringf( "%u\n", _lowerBounds.size() ) );
+    queryFile->write( Stringf( "%u\n", _upperBounds.size() ) );
 
     // Number of Equations
     queryFile->write( Stringf( "%u\n", _equations.size() ) );
 
     // Number of constraints
     queryFile->write( Stringf( "%u", _plConstraints.size() ) );
 
-    printf("Number of variables: %u\n", _numberOfVariables);
-    printf("Number of bounds: %u\n", _lowerBounds.size());
-    printf("Number of equations: %u\n", _equations.size());
-    printf("Number of constraints: %u\n", _plConstraints.size());
+    printf( "Number of variables: %u\n", _numberOfVariables );
+    printf( "Number of lower bounds: %u\n", _lowerBounds.size() );
+    printf( "Number of upper bounds: %u\n", _upperBounds.size() );
+    printf( "Number of equations: %u\n", _equations.size() );
+    printf( "Number of constraints: %u\n", _plConstraints.size() );
+
+    // Input Variables
+    queryFile->write( Stringf( "\n%u", getNumInputVariables() ) );
+    unsigned i = 0;
+    for ( const auto &inVar : getInputVariables() )
+    {
+        queryFile->write( Stringf( "\n%u,%u", i, inVar ) );
+        ++i;
+    }
+
+    // Output Variables
+    queryFile->write( Stringf( "\n%u", getNumOutputVariables() ) );
+    i = 0;
+    for ( const auto &outVar : getOutputVariables() )
+    {
+        queryFile->write( Stringf( "\n%u,%u", i, outVar ) );
+        ++i;
+    }
```

Comment:
* To keep consistency of the file, add description (Number of input variables etc...)
* I would also assert that _i_ is the same as the getNumOutputVariables (otherwise the load method will be inconsistent) same thing for input

Reply:
Added the assertion and better descriptions

Reply:
Oh no I understand that my comment about the description was kind of stupid, sorry.
I thought that we write to the file: NumberOfVariables: XXX 
my mistake, thanks for adding the asserts.